### DISCUSSION TITLE
What are the steps to access/enable the TiVo Mini Remote using VLC?

### DISCUSSION THREAD

**Discussion by MrBiggestDaddy** [GerryDazoo/Slinger#460] [Q&A] [answered]
Hi,
I am new to the community and have been able to get my M2 Slingbox to display a picture using VLC (VideoLAN - 3.0.8 VLC Media Player) on my laptop but cannot figure out how to get the TiVo Mini remote to display, e.g., I cannot control my TiVo from VLC? Any guidance would be greatly appreciated!

Happy holidays!

**Comment by tom76017** [comment, marked as answer]
There are several attempts to do this, but for me, I use a separate app to control my TiVo Mini.  There is about 3 second delay since the Slinger on VLC is buffered and the direct TiVo solution is instant.

For an iPhone take a look at https://github.com/GerryDazoo/Slinger/discussions/30

For Windows/MAC take a look at https://github.com/GerryDazoo/Slinger/discussions/84

I thought they were easy to install and work consistently well.

**Reply by MrBiggestDaddy** [reply]
Thanks tom76017! So I have a PC laptop I clicked on the #84 link you provided. From what I understand by reading I should be able to just click on the "Network Remote.pyw" file I downloaded? When I do a cmd window opens real fast but goes away ...and no remote keys work in VLC? I have Python 3.11 64bit installed so I am not sure what else to do?

Thanks in advance for any step-by-step guidance you can share!

**Reply by tom76017** [reply]
I'm still running python 2.7, but the readme says v3.x should work.

This is not part of VLC or Slinger.  It is an app the you talk directly to your TiVo.  You must enable Network Remote Control on your TiVo.  There's a few more steps to enable it to work externally from your LAN.

![image](https://user-images.githubusercontent.com/26171888/209961675-edff52a8-b14a-4f6c-9ff3-c168df70fd86.png)

**Reply by MrBiggestDaddy** [reply]
Right, I did that step on my TiVo to enable/allow the network remote (I did it on my main TiVo box which automatically was set on the mini it looks like). Then I connected to my network media stream with VLC and and am watching my TiVo channel via VLC. Then I double clicked on the "Network Remote.pyw" file. That is when all I get is a quick flash of a window like the command prompt window but cannot see that anything happened and I see no remote app nor did the shortcut keys work in VLC to control the TiVo?

**Reply by tom76017** [reply]
When I click on **"Network Remote.pyw"** I get this.  The IP address is my Mini.  
![image](https://user-images.githubusercontent.com/26171888/209983602-02de99d0-411b-485e-90ca-f317513a4906.png)

After electing the IP I get this

![image](https://user-images.githubusercontent.com/26171888/209983706-552f5ebd-f47b-4ddc-b568-0827a0c00448.png)

**Reply by MrBiggestDaddy** [reply]
Thanks and yes I would expect that to come up for me however the initial window does not come up to let me choose my mini? I wonder what could be stopping it from displaying in my case? After setting your main TiVo to allow remote control access did you have to restart or reboot anything regarding the TiVos or your router? Do I need to open any specific port(s) to enable the remote control program to access my mini possibly?

**Reply by tom76017** [reply]
I'm not a python programmer, but try right clicking on setup-exe.py and run IDLE.  Then from IDLE, run it.  See if that is a clue.

**Reply by MrBiggestDaddy** [reply]
Thank you for all your help ...I appreciate you! I ended up uninstalling Python and re-installing as administrator and changed the default installation folder. All is good now ...maybe python was not installed correctly I think.

Thanks again and have a Happy New Year!

**Reply by tom76017** [reply]
I installed python 3.11 on my PC and I get the same blink and crash that you are getting.  So, I think that you have to have python 2.7 installed.  I'm running Slinger on a Linux box, so Python is not a problem on my PC.  Based on Gerry's README, you just have to use the **_slingbox_server.exe_** and a Python 3 is not required.  If you don't have other PY apps, then try uninstalling Python 3 and install Python 2.7.

I tried to just rename Python to python2.exe and python3.exe with a co-installation and it caused Windows target problems.

**Reply by MrBiggestDaddy** [reply]
Weird. Well Python 3.11 launches the remote now. It did set Windows environment variables and installed to "C:\Program Files\Python311". So now I launch the remote first, then I run Gerry's "RunSling.bat" which starts Slinger 
cd slingbox
slingbox_server.exe

...then I open VLC and start the network stream to my Tivo Mini. 

Just wondering if you have played much with any of the "config.ini" settings to get a better picture (or some other options)? Back when Slingbox was supported I believe the picture was somewhat better?

**Reply by tom76017** [reply]
`Best I've found is 720 after trying all resolutions.  I'm using a Mini A92 with component connection.

**_`
Resolution=16
VideoBandwidth=1984
VideoSmoothness=59
VideoSource = 1_**

![image](https://user-images.githubusercontent.com/26171888/210073042-1005653f-7d83-44f9-942c-f9ca8c4f0191.png)

**Reply by tom76017** [reply]
Also, I created a shortcut on my Win11 desktop and changed the target to include my TiVo Mini IP address. This opens the app without selecting the Mini's IP.

![image](https://user-images.githubusercontent.com/26171888/210090170-b56f3035-34c4-44d8-93bd-11b6024fea5d.png)

**Reply by MrBiggestDaddy** [reply]
Thanks! I will give both those suggestions a try on my setup too. I am using the same ...component connection but the TiVo Mini (model TCDA93000).

Sorry, 1 more question please? How do you get the config.ini settings to take effect in VLC? I changed the settings but the network stream appears unchanged?

![image](https://user-images.githubusercontent.com/121572674/210126044-d044ea8f-e78e-4e46-970c-e4469e0283f2.png)

And once I start the server the config changes appear ...

![image](https://user-images.githubusercontent.com/121572674/210126062-ccd5ae75-c633-4a63-b027-bd0eb9b0e013.png)

But the VLC media info did not change?

![image](https://user-images.githubusercontent.com/121572674/210126084-e13b7a1f-2a88-43db-9f99-18b443ef3aa5.png)

**Reply by tom76017** [reply]
Every restart of VLC re-reads the video setup, so a restart of Slinger is not necessary.  Your A93 is getting a better signal than my A92.  My 720 may be a limitation of my Spectrum feed.  I don't think that Slinger or the Slingbox will upgrade the signal.

**Reply by MrBiggestDaddy** [reply]
Okay thanks!

**Reply by sp222222** [reply]
you guys both using m1 slingboxes? I wish my vid rez was that good on ProHD. and 500

**Reply by MrBiggestDaddy** [reply]
My Sling box says it is an M2

**Reply by tom76017** [reply]
Mine is an M1 with a Mini A92.  I also have a 500 on the shelf, but it gave me HDCP errors with a HDMI connection.  I tried both HDMI-2-Compoent connection and a HDMI splitter, but the 720 was the most acceptable connection.  I used an external connection to PC running VLC at a remote location driving a Samsung HDTV.  I think the Samsung upgraded the signal.  I was quite satisfied.  

Thanks Gerry!
